### ISSUE TITLE
TextField onChange, only value is passed

### ISSUE THREAD

**Issue by milep** [Shopify/polaris#83]
# Issue summary

Could the event be passed to the onChange function instead of just value? Or am I supposed to create individual handler functions for each input? How should I handle case like [Handling Multiple Inputs](https://facebook.github.io/react/docs/forms.html#handling-multiple-inputs) when I have a form with multiple dynamically created inputs?

**Comment by buggy** [comment]
@milep The easiest solution for now is to inline your handler

```javascript
onChange={val => this.setState({ name: val })}
```

or, if you're using a handler that's passed into your component as a property then something like

```javascript
onChange={val => props.handler("name", val)}
```

**Comment by milep** [comment]
@buggy When I try your proposed solution, the TextField input stops updating. It doesn't accept any input.
Here is an example:
```
import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import {Page, Card, FormLayout, TextField} from '@shopify/polaris';

document.addEventListener('DOMContentLoaded', () => {

  class Demo extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        test: 'hello'
      };
    }

    render() {
      return (
        <FormLayout>
          <TextField label="onChange test" name="test" onChange={(val) => {
              console.log(val);
              this.setState({test: val});
              }
            }/>
        </FormLayout>
      );
    }
  }
  
  ReactDOM.render(
    <Page title="Example app">
      <Card sectioned>
        <form>
          <Demo />
        </form>
      </Card>
    </Page>,
    document.getElementById('root')
  );
})
```
`console.log(val)` outputs the inserted character, but the input value doesn't change.

There is similar issue on https://polaris.shopify.com/components/forms/form-layout#examples, form fields do not accept input.

**Comment by lemonmade** [comment]
@milep your example is missing `value={this.state.test}` in the `TextField` component. I will put up an issue on our internal repo to discuss adding the event, probably as the second parameter, for all form-related change events. We did consciously choose to pass only the new value in the change event for our initial release, though: it's what you want most of the time, and having the raw event makes it tempting to circumvent React's model and do stuff on the DOM node directly.

**Comment by milep** [comment]
@lemonmade Thanks, `value={this.state.test}` solved it. I missed it from the examples.

I'm new to React, so this quote from documentation got me a bit worried:

> The problem with this syntax is that a different callback is created each time the LoggingButton renders. In most cases, this is fine. However, if this callback is passed as a prop to lower components, those components might do an extra re-rendering. We generally recommend binding in the constructor or using the property initializer syntax, to avoid this sort of performance problem.

**Comment by lemonmade** [comment]
Yeah, that caution from React is a reasonable one, although it's a bit of a micro-optimization that should only be used in cases where you know there is a performance hot path. There are also some other ways of solving this (for instance, we have a memoize function that returns a function to update values, so we only ever get one function for a given field name). We'll still look at adding the event as a parameter so we can get the best of both worlds, though.

**Comment by lemonmade** [comment]
Should be fixed by v1.1.0 (we now pass in the ID of the input as the second argument to `onChange` for all form components)

**Comment by solutionwin2020** [comment]
There are two ways to do this 

1st one is to pass name and value to a function
```
handleChange(name,event)
      {
        this.setState({[name]:event});
      }
onChange={event => this.handleChange('storeEmail',event)} value={this.state.storeEmail} 
```

and seconone is discussed above to use it inline 

`onChange={(event) => {this.setState({storeName:event})}}  value={this.state.storeName} `
